HouseRules_Essentials/Rules/RegroupAlliesRule.cs

```diff
@@ -0,0 +1,97 @@
+﻿namespace HouseRules.Essentials.Rules
+{
+    using System.Collections.Generic;
+    using Boardgame;
+    using Boardgame.BoardEntities;
+    using Boardgame.GameplayEffects;
+    using DataKeys;
+    using HarmonyLib;
+    using HouseRules.Types;
+
+    public sealed class RegroupAlliesRule : Rule, IConfigWritable<int>, IPatchable, IMultiplayerSafe
+    {
+        public override string Description => "Allies are included in Regroup.";
+
+        private static int _maxNum;
+        private static bool _isActivated;
+
+        private readonly int _adjustments;
+
+        private enum TeleportTargetPiece
+        {
+            Attacker,
+            Defenders,
+            MyTeam,
+        }
+
+        public RegroupAlliesRule(int adjustments)
+        {
+            _adjustments = adjustments;
+        }
+
+        public int GetConfigObject() => _adjustments;
+
+        protected override void OnActivate(GameContext gameContext)
+        {
+            _maxNum = _adjustments;
+            _isActivated = true;
+        }
+
+        protected override void OnDeactivate(GameContext gameContext) => _isActivated = false;
+
+        private static void Patch(Harmony harmony)
+        {
+            harmony.Patch(
+                original: AccessTools.Method(typeof(Teleport), "GetPiecesToTeleport"),
+                prefix: new HarmonyMethod(
+                    typeof(RegroupAlliesRule),
+                    nameof(Teleport_GetPiecesToTeleport_Prefix)));
+        }
+
+        private static bool Teleport_GetPiecesToTeleport_Prefix(ref List<Target> __result, TeleportTargetPiece mode, int maxNum, AbilityContext abilityContext, PieceAndTurnController pieceAndTurnController)
+        {
+            if (!_isActivated)
+            {
+                return true;
+            }
+
+            List<Target> list = new List<Target>(maxNum);
+            if (mode == TeleportTargetPiece.Attacker || mode == TeleportTargetPiece.Defenders)
+            {
+                return true;
+            }
+            else if (mode == TeleportTargetPiece.MyTeam)
+            {
```

Comment:
Can remove `else if` to reduce nesting (since it'll absolutely be the case if execution gets to this part).

A huge benefit to the [guard clause](https://refactoring.guru/replace-nested-conditional-with-guard-clauses) that you introduced above.